File: leetsync/3447-clear-digits/clear-digits.java
class Solution {
    public String clearDigits(String s) {
        int n = s.length();
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < n; i++) {
            char ch = s.charAt(i);
            if (Character.isDigit(ch)) {
                if (result.length() > 0) {
                    result.deleteCharAt(result.length() - 1);
                }
            } else {
                result.append(ch);
            }
        }
        return result.toString();
    }
}

File: leetsync/3447-clear-digits/clear-digits.go
func clearDigits(s string) string {
	n := len(s)
	result := ""
	for i := 0; i < n; i++ {
		ch := s[i]
		if ch >= '0' && ch <= '9' {
			result = result[:len(result)-1]
		} else {
			result = result + string(ch)
		}
	}
	return result
}